contracts/IdentityV1.sol

```diff
@@ -0,0 +1,277 @@
+// Taken from https://github.com/OriginProtocol/identity-playground/blob/0c9ba5c008d410e1ca82a2b1eed15705db49af0f/contracts/KeyHolder.sol
+// with some modifitcatioms
+
+pragma solidity ^0.4.22;
+
+import "./ERC725.sol";
+
+contract ERC20Basic {
+    function balanceOf(address _who) public constant returns (uint256);
+    function transfer(address _to, uint256 _value) public returns (bool);
+}
+
+contract IdentityV1 is ERC725 {
+
+    uint256 constant LOGIN_ONLY_KEY = 10;
+    uint256 constant FUNDS_MANAGEMENT = 11;
+
+    uint256 executionNonce;
+
+    struct Execution {
+        address to;
+        uint256 value;
+        bytes data;
+        bool approved;
+        bool executed;
+    }
+
+    mapping (bytes32 => Key) keys;
+    mapping (uint256 => bytes32[]) keysByPurpose;
+    mapping (uint256 => Execution) executions;
+
+    event ExecutionFailed(uint256 indexed executionId, address indexed to, uint256 indexed value, bytes data);
+
+    constructor() public {
+        bytes32 _key = keccak256(msg.sender);
+        keys[_key].key = _key;
+        keys[_key].purpose = [MANAGEMENT_KEY];
+        keys[_key].keyType = 1;
+        keysByPurpose[MANAGEMENT_KEY].push(_key);
+        emit KeyAdded(_key, MANAGEMENT_KEY, 1);
+    }
+
+    function getKey(bytes32 _key)
+        public
+        view
+        returns(uint256[] purpose, uint256 keyType, bytes32 key)
+    {
+        return (keys[_key].purpose, keys[_key].keyType, keys[_key].key);
+    }
+
+    function getKeyPurpose(bytes32 _key)
+        public
+        view
+        returns(uint256[] purpose)
+    {
+        return (keys[_key].purpose);
+    }
+
+    function getKeysByPurpose(uint256 _purpose)
+        public
+        view
+        returns(bytes32[] _keys)
+    {
+        return keysByPurpose[_purpose];
+    }
+
+    modifier onlyManagement() {
+        require(keyHasPurpose(keccak256(msg.sender), MANAGEMENT_KEY), "Sender does not have management key");
+        _;
+    }
+
+    modifier onlyAction() {
+        require(keyHasPurpose(keccak256(msg.sender), ACTION_KEY), "Sender does not have action key");
+        _;
+    }
+
+    modifier onlyFundsManagement() {
+        require(keyHasPurpose(keccak256(msg.sender), FUNDS_MANAGEMENT), "Sender does not have funds key");
+        _;
+    }
+
+    function addKey(bytes32 _key, uint256 _purpose, uint256 _type)
+        public
+        onlyManagement
+        returns (bool success)
+    {
+        if (keyHasPurpose(_key, _purpose)) {
+            return true;
+        }
+
+        keys[_key].key = _key;
+        keys[_key].purpose.push(_purpose);
+        keys[_key].keyType = _type;
```

Comment:
I am assuming that the purpose can be anything and its still valid but doesnt _type needs to be one of the ones specified

Reply:
Actually these are not specified by the standard at all... Just some suggestions

Reply:
oh okay